### PR TITLE
feat: added utils-go package and implemented for main.go files

### PR DESCRIPTION
**Pull request by commoddity**
Added the utils-go shared package that we use for repeated utility logic between repos.

In this case, just added the environment package that we use across all other services and updated the format of both `main.go` files. This has a few benefits:

1. consistency across all our backend repos - all the `main.go` files have a familiar format and layout now, in terms of how they handle environment variables.
2. includes the `MustGet...` funcs that throw an error if a required environment variable is not present in the environment as soon as the app starts
3. allows for running the relay meter locally and connecting to local influx & postgres instances which makes development easier

### CODE REVIEW

**Review by pablocampogo — DISMISSED** [review]

File: cmd/apiserver/main.go [hunk 1]
```diff
@@ -12,91 +12,43 @@ import (
...
-		*o.value = value
-	}
-
-	backendUrl := os.Getenv(ENV_BACKEND_API_URL)
-	if backendUrl == "" {
-		return options, fmt.Errorf("Missing required environment variable: %s", ENV_BACKEND_API_URL)
-	}
-	options.backendApiUrl = backendUrl
+var (
+	postgresUser     = environment.MustGetString("POSTGRES_USER")
```
Comment (adshmh) [inline cmd/apiserver/main.go]: Moving these to global level is the reverse of what we need to do, i.e. each variable should only have the absolute minimum scope that is required for it to be useful.

File: cmd/apiserver/main.go [hunk 2]
```diff
@@ -12,91 +12,43 @@ import (
...
-	if token == "" {
-		return options, fmt.Errorf("Missing required environment variable: %s", ENV_BACKEND_API_TOKEN)
-	}
-	options.backendApiToken = token
+	backendAPIURL   = environment.MustGetString("BACKEND_API_URL")
+	backendAPIToken = environment.MustGetString("BACKEND_API_TOKEN")
 
-	return options, nil
-}
+	loadInterval            = int(environment.GetInt64("LOAD_INTERVAL_SECONDS", 30))
```
Comment (adshmh) [inline cmd/apiserver/main.go]: Removing the options struct in favor of many variables causes scattering of these vars without being part of a single Options/Config struct.

File: cmd/apiserver/main.go [hunk 3]
```diff
@@ -12,91 +12,43 @@ import (
...
 )
 
-const (
-	LOAD_INTERVAL_DEFAULT_SECONDS      = 30
-	DAILY_METRICS_TTL_DEFAULT_SECONDS  = 120
-	TODAYS_METRICS_TTL_DEFAULT_SECONDS = 60
-	MAX_ARCHIVE_AGE_DEFAULT_DAYS       = 30
-	SERVER_PORT_DEFAULT                = 9898
-
-	ENV_LOAD_INTERVAL_SECONDS      = "LOAD_INTERVAL_SECONDS"
```
Comment (adshmh) [inline cmd/apiserver/main.go:29]: IMO these are useful constants that show clearly the expected settings and their default values, rather than allowing them to be hard-coded strings elsewhere.

File: cmd/apiserver/main.go (hunk 2 above)
Comment (adshmh) [reply cmd/apiserver/main.go]: Also, the default values should stay as they were, i.e. constants declared at the top, and used here, rather than scattering these default values over the code: keeping them as constant at the top makes it really easy to locate the config options and their default values, rather than having to search the code for it.

File: cmd/apiserver/main.go [hunk 4]
```diff
@@ -12,91 +12,43 @@ import (
...
-		envVar       string
-	}{
-		{value: &options.loadInterval, defaultValue: LOAD_INTERVAL_DEFAULT_SECONDS, envVar: ENV_LOAD_INTERVAL_SECONDS},
-		{value: &options.dailyMetricsTTLSeconds, defaultValue: DAILY_METRICS_TTL_DEFAULT_SECONDS, envVar: ENV_DAILY_METRICS_TTL_SECONDS},
-		{value: &options.todaysMetricsTTLSeconds, defaultValue: TODAYS_METRICS_TTL_DEFAULT_SECONDS, envVar: ENV_TODAYS_METRICS_TTL_SECONDS},
-		{value: &options.maxPastDays, defaultValue: MAX_ARCHIVE_AGE_DEFAULT_DAYS, envVar: ENV_MAX_ARCHIVE_AGE_DAYS},
-		{value: &options.port, defaultValue: SERVER_PORT_DEFAULT, envVar: ENV_SERVER_PORT},
-	}
-
-	for _, o := range optsItems {
```
Comment (adshmh) [inline cmd/apiserver/main.go:63]: I'm OK with using an existing `utils-go` package from another repository if that helps reduce duplicate code. But the general pattern used here can still be kept: i.e. declaring all the options and their defaults, and using the lib/package to load them.

File: cmd/apiserver/main.go [hunk 5]
```diff
@@ -12,91 +12,43 @@ import (
...
+	todaysMetricsTTLSeconds = int(environment.GetInt64("TODAYS_METRICS_TTL_SECONDS", 60))
+	maxPastDays             = int(environment.GetInt64("MAX_ARCHIVE_AGE", 30))
+	port                    = int(environment.GetInt64("API_SERVER_PORT", 9898))
+)
 
 type backendProvider struct {
 	db.PostgresClient
-	backendApiUrl   string
-	backendApiToken string
+	backendAPIURL   string
```
Comment (adshmh) [inline cmd/apiserver/main.go]: This makes sense to do IMO. But it does not belong in this PR.
A separate PR that handles styling/naming would be much more readable.

File: cmd/apiserver/main.go [hunk 6]
```diff
@@ -187,13 +139,12 @@ func (b *backendProvider) LoadBalancers() ([]*repository.LoadBalancer, error) {
 func main() {
 	log := logger.New()
 
-	options, err := gatherOptions()
```
Comment (adshmh) [inline cmd/apiserver/main.go:190]: Why do we need to remove this function? It enables unit testing: performing the options collection directly in main results in code that is more challenging to test.

File: cmd/apiserver/main.go [hunk 7]
```diff
@@ -202,23 +153,22 @@ func main() {
...
-		backendApiToken: options.backendApiToken,
+		backendAPIURL:   backendAPIURL,
+		backendAPIToken: backendAPIToken,
 	}
 	meter := api.NewRelayMeter(&backend, log, meterOptions)
 	http.HandleFunc("/", api.GetHttpServer(meter, log))
 
 	log.Info("Starting the apiserver...")
-	http.ListenAndServe(fmt.Sprintf(":%d", options.port), nil)
+	http.ListenAndServe(fmt.Sprintf(":%d", port), nil)
```
Comment (adshmh) [inline cmd/apiserver/main.go]: Same as previous comments re options struct: We should not eliminate logical grouping of these settings (i.e. `options` struct) to get a group of variables that are not related to each other in code at all, but are actually all part of the configuration of the service.

File: cmd/collector/main.go [hunk 8]
```diff
@@ -6,71 +6,54 @@ import (
...
+	influxDBBucketCurrent = environment.MustGetString("INFLUXDB_BUCKET_CURRENT")
+
+	postgresUser     = environment.MustGetString("POSTGRES_USER")
+	postgresPassword = environment.MustGetString("POSTGRES_PASSWORD")
+	postgresDB       = environment.MustGetString("POSTGRES_DB")
+	postgresHost     = environment.MustGetString("POSTGRES_HOST")
+
+	collectionInterval = int(environment.GetInt64("COLLECTION_INTERVAL_SECONDS", 300))
+	reportingInterval  = int(environment.GetInt64("REPORT_INTERVAL_SECONDS", 30))
+	maxArchiveAgeDays  = int(environment.GetInt64("MAX_ARCHIVE_AGE", 30))
```
Comment (adshmh) [inline cmd/collector/main.go]: Same as previous comments: OK to use a separate package/lib to handle the env. loading.
However, this collection should be:
- using constant declared at the top for readability
- using options struct to group them, for readability and making the code easier to test.
- performed in its own function/method to ensure we can cover it with unit tests.

File: cmd/collector/main.go [hunk 9]
```diff
@@ -6,71 +6,54 @@ import (
...
-	return options{
-		collectionInterval: collectionInterval,
-		reportingInterval:  reportingInterval,
-		maxArchiveAgeDays:  maxArchiveAge,
-	}, nil
-}
-
 // TODO: need a /health endpoint
 func main() {
-	influxOptions := cmd.GatherInfluxOptions()
```
Comment (adshmh) [inline cmd/collector/main.go]: We should not remove this method or the one on the next line, which provide for a readable and easily testable building of the configuration options. 

Using the new shared lib/package for env. loading can still be performed within these methods.

File: cmd/options.go [hunk 10]
```diff
@@ -1,64 +0,0 @@
...
-	POSTGRES_HOST     = "POSTGRES_HOST"
-)
-
-type options struct {
-	db.InfluxDBOptions
-	db.PostgresOptions
-}
-
-func GatherInfluxOptions() db.InfluxDBOptions {
-	return db.InfluxDBOptions{
```
Comment (adshmh) [inline cmd/options.go:29]: Same as previous comments re keeping items that belong together in the same method and struct.

File: cmd/options.go [hunk 11]
```diff
@@ -1,64 +0,0 @@
...
-		User:     os.Getenv(POSTGRES_USER),
-		Password: os.Getenv(POSTGRES_PASSWORD),
-		Host:     os.Getenv(POSTGRES_HOST),
-		DB:       os.Getenv(POSTGRES_DB),
-	}
-}
-
-// TODO: flags package makes this a lot easier, but env. variables are better suited for k8s deployments
-//	TODO: see if there is a package so we can skip writing env. processing code
-func GetIntFromEnv(envVarName string, defaultValue int) (int, error) {
```
Comment (adshmh) [inline cmd/options.go:49]: Thank you for addressing the TODO comment!
Letting an existing package handle the env. logic makes perfect sense.

**Review by adshmh — CHANGES_REQUESTED** [review]
Thank you for the PR!
I suggest choosing a better title: e.g. use a package for environment handling logic.

**Review by adshmh — APPROVED** [review]
Thank you for addressing the review comments.
LGTM.
